### ISSUE TITLE
Make temporal specification more flexible in GetObsRequest

### ISSUE THREAD

**Issue by jo-asplin-met-no** [EURODEO/e-soh#49]
In `GetObsRequest` we currently use a combination of `temporal_mode` and `temporal_interval` to specify one of the 
following two cases of filtering wrt. time (i.e. this filter is essentially applied on time series that match other filters like spatial search and attribute string matching):

1. get all observations in a specified interval (`temporal_mode` == 'interval' and `temporal_interval` is specified)
2. get the single latest observation in the entire buffer (last 24 hours) (`temporal_mode` == 'latest' and `temporal_interval` is ignored (i.e. can be set or not))

To make the temporal specification more flexible we propose to replace `temporal_mode` with a boolean flag `temporal_latest` to support the following four cases:

1. get all observations in the entire buffer (last 24 hours) (`temporal_latest` == _false_ and `temporal_interval` is not specified)
2. get all observations in a specified interval (`temporal_latest` == _false_ and `temporal_interval` is specified)
3. get the single latest observation in the entire buffer (last 24 hours) (`temporal_latest` == _true_ and `temporal_interval` is not specified)
4. get the single latest observation in a specified interval (`temporal_latest` == _true_ and `temporal_interval` is specified)

In other words:

<html>
<table>
<tr><th>temporal_latest</th><th>temporal_interval specified</th><th>description</th></tr>
<tr><td>false</td><td>no</td><td>get all observations in the entire buffer (last 24 hours)</td></tr>
<tr><td>false</td><td>yes</td><td>get all observations in a specified interval</td></tr>
<tr><td>true</td><td>no</td><td>get the single latest observation in the entire buffer (last 24 hours)</td></tr>
<tr><td>true</td><td>yes</td><td>get the single latest observation in a specified interval</td></tr>
</table>
</html>
